cashu/mint/dlc.py

```diff
@@ -0,0 +1,98 @@
+from .ledger import Ledger
+from ..core.models import PostDlcRegistrationRequest, PostDlcRegistrationResponse
+from ..core.base import DlcBadInput, DlcFundingProof, Proof, DLCWitness
+from ..core.secret import Secret, SecretKind
+from ..core.crypto.dlc import list_hash, merkle_verify
+from ..core.errors import TransactionError
+from ..core.nuts import DLC_NUT
+
+
+from hashlib import sha256
+from loguru import logger
+from typing import List, Dict, Optional
+
+class LedgerDLC(Ledger):
+
+    async def _verify_dlc_input_spending_conditions(self, dlc_root: str, p: Proof) -> bool:
+        if not p.witness:
+            return False
+        witness = DLCWitness.from_witness(p.witness)
+        leaf_secret = Secret.deserialize(witness.leaf_secret)
+        secret = Secret.deserialize(p.secret)
+        # Verify secret is of kind SCT
+        if secret.kind != SecretKind.SCT.value:
+            return False
+        # Verify leaf_secret is of kind DLC
+        if leaf_secret.kind != SecretKind.DLC.value:
+            return False
+        # Verify dlc_root is the one referenced in the secret
+        if leaf_secret.data != dlc_root:
+            return False
+        # (Optional?) Verify inclusion of leaf_secret in the SCT root hash
```

Comment:
For SCTs in general, inclusion of the `leaf_secret` in the SCT root hash is definitely _not_ optional